cartography/intel/aws/iam.py

```diff
@@ -364,38 +385,198 @@ def load_user_access_keys(neo4j_session, user_access_keys, aws_update_tag):
+    """
+    neo4j_session.run(
+        injest_policy,
+        PolicyId=policy_id,
+        PolicyName=policy_name,
+        PolicyType=policy_type,
+        PrincipalArn=principal_arn,
+        aws_update_tag=aws_update_tag,
+    ).consume()
+
+
+@timeit
+def load_policy_statements(neo4j_session, policy_id, policy_name, statements, aws_update_tag):
+    injest_policy_statement = """
+        MATCH (policy:AWSPolicy{id: {PolicyId}})
+        WITH policy
+        UNWIND {Statements} as statement_data
+        MERGE (statement:AWSPolicyStatement{id: statement_data.id})
+        SET
+        statement.effect = statement_data.Effect,
+        statement.action = statement_data.Action,
+        statement.notaction = statement_data.NotAction,
+        statement.resource = statement_data.Resource,
+        statement.notresource = statement_data.NotResource,
+        statement.condition = statement_data.Condition,
+        statement.sid = statement_data.Sid,
+        statement.lastupdated = {aws_update_tag}
+        MERGE (policy)-[r:STATEMENT]->(statement)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {aws_update_tag}
+        """
+    neo4j_session.run(
+        injest_policy_statement,
+        PolicyId=policy_id,
+        PolicyName=policy_name,
+        Statements=statements,
+        aws_update_tag=aws_update_tag,
+    ).consume()
+
+
+@timeit
+def load_policy_data(neo4j_session, policy_map, aws_update_tag):
+    for principal_arn, policy_list in policy_map.items():
+        logger.debug(f"Syncing IAM inline policies for principal {principal_arn}")
+        for policy in policy_list:
+            name = policy.name
+            policy_id = f"{principal_arn}/inline_policy/{name}"
+            try:
+                statements = _transform_policy_statements(
+                    policy.policy_document["Statement"], policy_id,
+                )
+                load_policy(neo4j_session, policy_id, name, "inline", principal_arn, aws_update_tag)
+                load_policy_statements(neo4j_session, policy_id, name, statements, aws_update_tag)
+            except ClientError:
+                # Avoid crashing the sync
+                logger.warning("inline policy {name} failed with ClientError; skipping.")
+                continue
+
+
+@timeit
+def load_managed_policy_data(neo4j_session, policy_map, aws_update_tag):
+    for principal_arn, policy_list in policy_map.items():
+        for policy in policy_list:
+            name = policy.policy_name
+            policy_id = policy.arn
+            try:
+                statements = _transform_policy_statements(
+                    policy.default_version.document["Statement"], policy_id,
+                )
+                load_policy(neo4j_session, policy_id, name, "managed", principal_arn, aws_update_tag)
+                load_policy_statements(neo4j_session, policy_id, name, statements, aws_update_tag)
+            except ClientError:
```

Comment:
Same comment here: will this ever be thrown? There are no boto3 calls in this function.

Reply:
The resource clients can still throw client errors when accessing related items.